readium/navigator/src/main/java/org/readium/r2/navigator/media2/MediaNavigator.kt

```diff
@@ -0,0 +1,99 @@
+package org.readium.r2.navigator.media2
+
+import kotlinx.coroutines.flow.StateFlow
+import org.readium.r2.navigator.ExperimentalAudiobook
+import org.readium.r2.navigator.Navigator
+import org.readium.r2.shared.publication.Link
+import org.readium.r2.shared.publication.Locator
+import org.readium.r2.shared.util.Try
+import kotlin.time.Duration
+import kotlin.time.ExperimentalTime
+
+/**
+ * A navigator rendering an audio or video publication.
+ */
+@ExperimentalAudiobook
+@OptIn(ExperimentalTime::class)
+interface MediaNavigator : Navigator {
```

Comment:
I'm a bit on the fence for this interface. It doesn't seem to bring much compared to the existing `MediaNavigator` interface and the whole purpose was to have a single interface for various media-based navigators. Having one interface per implementation defeats the purpose.

I get that it adds coroutine support, but I'd rather update the existing interface and media1 implementation to use coroutines (even if it's only in the API signatures and not really in the implementation). It still has the `@ExperimentalAudiobook` annotation so it's fine to make breaking changes.

Reply:
I didn't think the two interfaces should coexist.  We are not in a case with two different media navigators, but with one supposed to replace the other one (still experimental). If breaking the interface is not a problem, let's just remove the previous navigator.

Reply:
Even though the media1 navigator is still marked as experimental, I still want to have a migration period before removing it entirely. That would also allow to merge this PR in even if it's not yet fully on par with the media1 navigator feature-wise and not yet battle-tested.

For the `MediaNavigator` interface, I'm considering that maybe we should just remove it. Contrarily to the other navigator interfaces, we will probably have only a single implementation in Readium, so it's not very useful. I suggest that we wait until we have another implementation (e.g. for video publications) in Readium or an open third-party audio navigator before extracting a shared interface.

Reply:
Agreed.